### PR TITLE
Add timeout support and more baud rates

### PR DESCRIPTION
**Pull request by jacobsparts**
timeout support via uselect
inWaiting method doesn't work for me, otherwise that could be used instead of uselect

### CODE REVIEW

File: serial.py [hunk 1]
```diff
@@ -22,12 +23,17 @@ class SerialDisconnectException(SerialException):
 
 class Serial:
 
-    BAUD_MAP = { 9600: termios.B9600, 115200: termios.B115200 }
+    BAUD_MAP = {
```
Comment (pfalcon) [inline serial.py:26]: Can you please split baud rate additions to a separate commit, as they're not related to the timeout changes?

File: serial.py [hunk 2]
```diff
@@ -22,12 +23,17 @@ class SerialDisconnectException(SerialException):
...
+        9600: termios.B9600,
+        19200: 14,
```
Comment (pfalcon) [inline serial.py:28]: Can you add comment what is 14/where it comes from?

File: serial.py [hunk 3]
```diff
@@ -22,12 +23,17 @@ class SerialDisconnectException(SerialException):
...
+        57600: termios.B57600,
+        115200: termios.B115200
+    }
 
     def __init__(self, port, baudrate, timeout=None, **kwargs):
-        assert timeout is None, "Only no-timeout mode is supported"
         self.port = port
         self.baudrate = baudrate
+        self.timeout = None if timeout is None else timeout * 1000
```
Comment (pfalcon) [inline serial.py]: Did you actually test this? In MicroPython, poll() argument for "no timeout" is -1.

**Review by pfalcon — CHANGES_REQUESTED** [review]
Looks good and useful! Few comments how to make it perfect. Let me know if you would go for the changes (if not, I may do that instead, as time permits).

File: serial.py (hunk 1 above)
Comment (jacobsparts) [reply serial.py:26]: done

File: serial.py (hunk 2 above)
Comment (jacobsparts) [reply serial.py:28]: termios baud rate constants have integer values which ultimately refer to positions in the tty baud rate table in the [linux kernel](https://github.com/torvalds/linux/blob/894025f24bd028942da3e602b87d9f7223109b14/drivers/tty/tty_baudrate.c).  Micropython's termios does not define all of the baud rate constants, but they can be referred to by their integer values.

File: serial.py (hunk 3 above)
Comment (jacobsparts) [reply serial.py]: Yes, I did test it.  None and -1 produce the same behavior, but we can use -1 instead.

File: serial.py (hunk 2 above)
Comment (pfalcon) [reply serial.py:28]: Thanks for the details, I actually added it as an inline comment too.

**Comment by pfalcon** [comment]
Thanks for the refactors! I did few more cleanups and merged/released new version to PyPI.
